linchpin/examples/controller-provisioning-linchpin/README.md

```diff
@@ -1,32 +1,42 @@
+The controller user is `kstest`. Ssh key for the user access is configured by `kstest_controller_user_authorized_key` variable which can be set in inventory via [kstests-controller-provision.sh](kstests-controller-provision.sh).
 
-In this example the variables can be configured in the provisioning script [kstests-controller-provision.sh](kstests-controller-provision.sh) by updating the `[kstest-controller:vars]` section of the generated inventory.
+Assuming the controller will be used to provision runners in cloud, the cloud credentials need to be passed to the controller via a configuration file. Default path to the file is configured by `cloud_config_file` variable in [kstest-controller/defaults/main.yml](../../../ansible/roles/kstest-controller/defaults/main.yml). In this example it can be overriden in the inventory via [kstests-controller-provision.sh](kstests-controller-provision.sh).
 
-## Provision and deploy:
+It may be useful to pass also an ssh keypair to be further used to the controller. The keys to be passed to the controller user can be configured by the playbook variables `private_keys_to_upload`, `public_keys_to_upload` that can be set for example in the inventory via [kstests-controller-provision.sh](kstests-controller-provision.sh).
 
-Copy [PinFile.controller](linchpin/PinFile.controller) containing resources definition of *controller* host to the `linchpin` folder.
+### Provision and deploy
 
-Copy [kstests-controller-provision.sh](kstests-controller-provision.sh) in `kickstart-tests` repo root and set the path to the private key in `<PRIVATE_KEY_PATH>`.
-Also set credentials to be passed to the *controller* in `<PUBLIC_KEY_TO_UPLOAD_PATH>`, `<PRIVATE_KEY_TO_UPLOAD_PATH>`, `<CLOUD_CONFIG_FILE>` (or comment out the lines if you don't intend to use them on the controller).
+- Update the deployment key name in [PinFile.controller](PinFile.controller) variable `keypair` to the keypair you want to use.
 
-Run [kstests-controller-provision.sh](kstests-controller-provision.sh) in `kickstart-tests` repo root.
+- Copy [kstests-controller-provision.sh](kstests-controller-provision.sh) in `kickstart-tests` repo root and configure the inventory:
+    - Replace the `<PRIVATE_KEY_PATH>` with path to the private key of the cloud keypair used above (if it is not your default key).
+    - Replace the `<PUBLIC_KEY_PATH>` with ssh public key for controller user (`kstest`) access.
+    - Replace the `<CLOUD_CONFIG_FILE>` with path to cloud configuration file and uncomment the variable or place the file to the default location as configured in [kstest-controller/defaults/main.yml](../../../ansible/roles/kstest-controller/defaults/main.yml).
```

Comment:
I think it should be ``..path to *the* cloud configuration file...``

Reply:
Fixed